Updates to enable live testing in sovereign clouds for multiple services

zedy-wj opened the pull request:
We are running live Tests against other clouds like US Gov and Azure China Cloud. The goal is to check whether new azure sdk package work with other clouds or not.

Using this RR to trace and fix issues we encounter in the tests. Will close it once we finish all the tests and send PR per services.

Updated list:
Text analytics
Core
Extensions
Event hubs
Identity
key vault
Search
Storage account
Form recognizer
Service bus
Event grid
Tables

zedy-wj committed:
Updates to enable live testing in sovereign clouds for multiple services

zedy-wj committed:
little update

zedy-wj committed:
Merge branch 'master' into wenjyu/sovereignCloud

jsquire commented on sdk/servicebus/tests.data.yml:
```diff
@@ -7,5 +7,5 @@ extends:
     SDKType: data
     TimeoutInMinutes: 240
-    Clouds: 'Public,Canary'
+    Clouds: 'Public,Preview,UsGov,China,Canary'
```
Do we want the track one library to be running in sovereign clouds?

//cc: @DorothySun216

jsquire replied:
After conversations with @benbp about the intent, we'll want to leave `Clouds` as-is and just ensure that `SupportedClouds` reflect the full set.   We want to limit sovereign clouds to just the weekly runs, not include them in the nightlies or manual runs.

jsquire commented on sdk/eventhub/tests.data.yml:
```diff
@@ -6,5 +6,5 @@ extends:
-    Clouds: 'Public,Canary'
\ No newline at end of file
+    TimeoutInMinutes: 240
+    Clouds: 'Public,Preview,UsGov,China,Canary' 
```
Do we want the track on tests running in sovereign clouds?    @serkantkaraca  should make the call.

jsquire commented on sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:
```diff
@@ -131,7 +131,7 @@ static TestEnvironment()
         ///   The URL of the Azure Authority host to be used for authentication. Recorded.
         /// </summary>
-        public string AuthorityHostUrl => GetRecordedOptionalVariable("AZURE_AUTHORITY_HOST");
+        public string AuthorityHostUrl => GetRecordedOptionalVariable("AZURE_AUTHORITY_HOST") ?? "https://login.microsoftonline.com/";
```
If I recall, @pakrym had some thoughts about including this as a lone default here...

pakrym replied:
The resource deployment script should provide this value: https://github.com/Azure/azure-sdk-tools/blob/154d977720f56e6816fb99e1e6ff6970321dda8a/eng/common/TestResources/New-TestResources.ps1#L492 we shouldn't hard code it here.

pakrym commented on sdk/storage/test-resources.json:
```diff
@@ -407,6 +446,14 @@
+        "type": "string",
+        "value": "[parameters('storageEndpointSuffix')]"
+    },
+    "AZURE_AUTHORITY_HOST": {
```
I don't think this is required. Net-TestResources already has these variables as the output.

zedy-wj committed:
Update tests.data.yml

zedy-wj committed:
Update some changes

zedy-wj committed:
Update some changes

zedy-wj committed:
Remove the hardcode.

zedy-wj committed:
Merge branch 'master' into wenjyu/sovereignCloud

zedy-wj committed:
If removing the hardcode, some identity jobs will fail.

maririos commented on sdk/textanalytics/tests.yml:
```diff
@@ -9,4 +9,8 @@ extends:
+        Location: 'centraluseuap'
+      UsGov:
+        SubscriptionConfiguration: $(sub-config-gov-test-resources-test)
+    Clouds: 'Public,Canary,UsGov'
```
I think some of the TA features are not enabled in UsGov. If the Healthcare tests start failing, then we shouldn't enable this could yet

pakrym replied:
We can consider [SkipInCloud()] attribute

christothes commented on sdk/tables/test-resources.json:
```diff
@@ -74,7 +74,7 @@
                 "locations": [
                     {
-                        "locationName": "[variables('location')]",
+                        "locationName": "[parameters('cosmosdblocation')]",
```
Where is `cosmosdblocation` defined? It doesn't seem to exist in New-TestResources.ps1

zedy-wj committed:
Update tests.data.yml

Remove preview from clouds

zedy-wj committed:
Update tests.yml

Remove Preview from clouds

zedy-wj committed:
Remove trailing whitespace.

zedy-wj committed:
Merge branch 'wenjyu/sovereignCloud' of https://github.com/zedy-wj/azure-sdk-for-net into wenjyu/sovereignCloud

zedy-wj committed:
Merge branch 'master' into wenjyu/sovereignCloud

zedy-wj committed:
Remove trailing whitespace.

zedy-wj committed:
Merge branch 'wenjyu/sovereignCloud' of https://github.com/zedy-wj/azure-sdk-for-net into wenjyu/sovereignCloud

zedy-wj committed:
Update test-resources.json

Remove changes

zedy-wj committed:
Update tests.yml

Add UsGov and China clouds, remove Preview cloud

zedy-wj committed:
Remove hardcoded value of AZURE_KEYVAULT_SCOPE.

zedy-wj committed:
Merge branch 'wenjyu/sovereignCloud' of https://github.com/zedy-wj/azure-sdk-for-net into wenjyu/sovereignCloud

zedy-wj committed:
Add hardcoded value back, remove it will cause some tests fail.

zedy-wj committed:
Update tests.yml

 Enable the HSM tests for usgov

zedy-wj committed:
Update tests.yml

Enable the HSM tests for usgov

zedy-wj committed:
Update tests.yml

The resource type Microsoft.KeyVault/managedHSMs was invalid in the namespace Microsoft.KeyVault for UsGov and China clouds

zedy-wj committed:
enable hsm tests in usgov

zedy-wj committed:
Update test-resources.json

Remove invalid changes

zedy-wj committed:
Set a default value for KeyvaultScope.

zedy-wj committed:
Exclude HSM tests for usgov and china.

zedy-wj committed:
Update identity clientId in sovereign cloud.

zedy-wj committed:
Updating parameter IdentityClientId to ClientId.

zedy-wj committed:
Remove invalid changes.

zedy-wj committed:
remove hardcode of DefaultCognitiveScope value in order to run job in UsGov and China clouds.

zedy-wj committed:
Merge branch 'master' into wenjyu/sovereignCloud

jsquire commented on sdk/servicebus/tests.yml:
```diff
@@ -7,5 +7,5 @@ extends:
     SDKType: client
     TimeoutInMinutes: 240
-    Clouds: 'Public,Canary'
+    Clouds: 'Public,Preview,UsGov,China,Canary'
```
After conversations with @benbp about the intent, we'll want to leave `Clouds` as-is and just ensure that `SupportedClouds` reflect the full set.   We want to limit sovereign clouds to just the weekly runs, not include them in the nightlies or manual runs.

jsquire commented:
I left comments around Service Bus specifically, but I believe this applies generally as well.   After conversations with @benbp about the intent, we'll want to leave `Clouds` as-is and just ensure that `SupportedClouds` reflect the full set.   We want to limit sovereign clouds to just the weekly runs, not include them in the nightlies or manual runs.

zedy-wj committed:
Update tests.data.yml

limit sovereign clouds to just the weekly runs

zedy-wj committed:
Update tests.yml

limit sovereign clouds to just the weekly runs

zedy-wj committed:
Update TextAnalytics codes

zedy-wj committed:
test text analytics in usgov

zedy-wj committed:
test form recognizer in China cloud

zedy-wj committed:
Enable form recognizer in China cloud

zedy-wj committed:
Remove hardcode in FormTrainingClientLiveTests.cs

zedy-wj commented on sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:
```diff
@@ -22,7 +22,7 @@ public class TextAnalyticsClient
         internal readonly ClientDiagnostics _clientDiagnostics;
         private readonly TextAnalyticsClientOptions _options;
-        private readonly string DefaultCognitiveScope = "https://cognitiveservices.azure.com/.default";
+        private readonly string DefaultCognitiveScope = TextAnalyticsClientOptions.getDefaultCognitiveScope(Environment.GetEnvironmentVariable("TENANT_ID"));
```
Look at the `TENANT_ID` was also my first choice, it looked good in the local test, but it could not get `EnvironmentVariable: TENANT_ID` in Pipeline. So i change the way and get `AZURE_AUTHORITY_HOST` for Form Recognizer. I will remove these changes in Text Analytics, because of `Healthcare` tests are failing in per clouds except Public.

maririos replied:
Hmm interesting. Getting `TENANT_ID` should always work as we always need it when running our test with AAD. See how our `TestEnvironment` does it [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L121).

As for Text Analytics, yeah, `Healthcare` only works on public. This might change in a couple of months though

zedy-wj replied:
I am strange for it. Was the value of `TENANT_ID` constant in every clouds? Maybe I set it's value wrong.
`Public : 72f988bf-86f1-41af-91ab-2d7cd011db47`
`UsGov  : 63296244-ce2c-46d8-bc36-3e558792fbee`
`China  : 3d0a72e2-8b06-4528-98df-1391c6f12c11`
